velox/expression/Expr.cpp

```diff
@@ -473,6 +473,26 @@ void Expr::evalFlatNoNulls(
     EvalCtx& context,
     VectorPtr& result,
     bool topLevel) {
+  if (deterministic_ && isMultiplyReferenced_ && !inputs_.empty()) {
```

Comment:
just curious, can you give an example of when inputs are empty and this flatNoNulls path is taken? it is for functions like pi()

Reply:
This happens for ConstantTypedExpr and FieldAccessTypedExpr. (It is an unfortunate gotcha that I'd love to remove at some point.)